### ISSUE TITLE
[Bug]: ml/adapters/xgboost_adapter.py — predict() crashes with IndexError on empty DataFrame, and silently skips column validation when feature_names are unavailable

### ISSUE THREAD

**Issue by prathamesh-lang** [Eshajha19/agri#1682]
Two bugs in XGBoostAdapter.predict():

1. Empty input IndexError — if an empty DataFrame is passed, model.predict() returns an empty array and prediction[0] raises an unhandled IndexError with no meaningful message.

```python
prediction = self.model.predict(input_data)
return float(prediction[0])  # IndexError if input_data is empty
```

Verified: numpy array index 0 is out of bounds for axis 0 with size 0.

2. Silent column validation bypass — self._feature_names is initialized as an empty list. For models that don't expose feature_names_in_ or get_booster().feature_names (e.g. some sklearn pipelines), _feature_names stays empty and the entire column validation block is skipped silently. A DataFrame with wrong or missing columns gets passed directly to the model, producing meaningless predictions with no error.

```python
if self._feature_names:  # silently skipped when list is empty
    missing = [c for c in self._feature_names if c not in input_data.columns]
    ...
```

File: ml/adapters/xgboost_adapter.py — predict() method, lines 44–59

Fix:

```python
def predict(self, input_data: pd.DataFrame) -> float:
    if self.model is None:
        raise ValueError("Model not loaded. Call load() first.")
    if len(input_data) == 0:
        raise ValueError("input_data is empty — cannot run inference on zero samples")
    if self._feature_names:
        missing = [c for c in self._feature_names if c not in input_data.columns]
        if missing:
            raise ValueError(f"Missing columns: {missing}")
        input_data = input_data[self._feature_names]
    prediction = self.model.predict(input_data)
    if len(prediction) == 0:
        raise ValueError("Model returned empty prediction output")
    return float(prediction[0])
```

Difficulty: Medium
Estimated Time: 3-4 hours

**Comment by Eshajha19** [comment]
Assigned @prathamesh-lang

**Comment by prathamesh-lang** [comment]
Resolves this issue via PR #1731 — added empty input guard and fixed silent column validation bypass in XGBoostAdapter.predict().
